Internal api fix eas 867

day0ops opened the pull request:
# Description

A number of small fixes for [867](https://github.com/solo-io/ext-auth-service/issues/867)
- internal API ext auth changes
- tweaks to the test http request builder

sheidkamp commented:
> LGTM. However, I don't know if it's a good practice to put in fields that we don't use yet. If people don't look at the doc, they might get confused. Once to take that field number, it's gone, you can still rename it but you cannot use it for anything else if you decide to change later when you actually implement it.

I don't think any of these will be unused. These are to support a solo-projects PR that I'd expect to see shortly. This time though, lets keep the API changes from merging until that PR is complete.

day0ops commented:
i think this is fairly well complete now.
frankly i dont think theres a need for them in this set of PRs so i removed them. given its internal i thought it doesnt need to be as polished as the public facing api hence why i added those additinal internal fields incase we need them.
